staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -489,6 +493,8 @@ func (cnc *CloudNodeController) syncNode(ctx context.Context, nodeName string) e
 			return err
 		}
 
+		removeNodeTaintDelay.Observe(time.Since(newNode.ObjectMeta.CreationTimestamp).Seconds())
+
 		// After adding, call UpdateNodeAddress to set the CloudProvider provided IPAddresses
 		// So that users do not see any significant delay in IP addresses being filled into the node
 		cnc.updateNodeAddress(ctx, newNode, instanceMetadata)
```

Comment:
I'd love to have one metrics for `UpdateNodeAddress`

Reply:
I kind of considered this, and decided it may not add much value beyond `removeNodeTaintDelay`, since it's just a call to the `Cloud` and another write to the API server. Are you thinking it'd be a similar value, time since node creation when the addresses are finalized?

Reply:
yes, it is important because the way of cloud provider assign IPs to nodes, specially ExternalIPs, it is interesting to know how long it take to get the IP assigned in the Node

Reply:
Fair enough, it's easy to add. I'll push a revision 👍

Reply:
I added 2 metrics for this:
1. The delay since node creation for the initial population of a node's Addresses.
2. The time taken to reconcile the addresses.